### PR TITLE
Fix minor typos

### PR DESCRIPTION
**Pull request by NathanBaulch**
Just thought I'd contribute some typo fixes that keep tripping up CI/CD checks in my projects. Nothing controversial (hopefully), just 174 simple fixes.

Use the following command to get a quick and dirty summary of the specific corrections made:
```shell
git diff HEAD^! --word-diff-regex='\w+' -U0 \
  | grep -E '\[\-.*\-\]\{\+.*\+\}' \
  | sed -r 's/.*\[\-(.*)\-\]\{\+(.*)\+\}.*/\1 \2/' \
  | sort | uniq -c | sort -n
```

FWIW, the top typos are:
* trimed → trimmed (37)
* substract → subtract (7)
* qualifed → qualified (7)
* extesion → extension (6)
* mising → missing (5)
* btye → byte (4)
* likey → likely (4)
* candicate → candidate (3)
* decriptor → descriptor (3)
* inherting → inheriting (3)
* colletion → collection (3)
* caluclated → calculated (3)
* unititialized → uninitialized (3)
* implemting → implementing (3)
* binrary → binary (3)
* descripor → descriptor (3)
* negitive → negative (3)

### CODE REVIEW

**Review by jskeet — COMMENTED** [review]
The C# changes look fine to me.

**Review by JasonLunn — COMMENTED** [review]
Ruby changes LGTM

**Comment by JasonLunn** [comment]
> Thanks for your pull request! It looks like this may be your first contribution to a Google open source project. Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).
> 
> View this [failed invocation](https://github.com/protocolbuffers/protobuf/pull/17682/checks?check_run_id=28194774429) of the CLA check for more information.
> 
> For the most up to date status, view the checks section at the bottom of the pull request.

@NathanBaulch Please sign the CLA as soon as possible.

**Comment by NathanBaulch** [comment]
> @NathanBaulch Please sign the CLA as soon as possible.

I signed it last week shortly after creating this PR @JasonLunn.
image

**Comment by NathanBaulch** [comment]
Sorry @JasonLunn, I force pushed to resolve a new conflict with `main` and lost the "safe for tests" status.

**Comment by NathanBaulch** [comment]
Bump!

**Review by JasonLunn — APPROVED** [review]

**Comment by tonyliaoss** [comment]
Hi Nathan, sorry for the late response, unfortunately it looks like there are new merge conflicts from some of the files going stale. Could you pull up to main and resolve the merge conflicts please?

File: src/google/protobuf/descriptor.proto [hunk 1]
```diff
@@ -80,7 +80,7 @@ enum Edition {
   EDITION_2024 = 1001;
 
   // Placeholder editions for testing feature resolution.  These should not be
-  // used or relyed on outside of tests.
```
Comment (mkruskal-google) [inline src/google/protobuf/descriptor.proto:83]: Unfortunately we can't accept any external contributions to this file

**Comment by tonyliaoss** [comment]
Hi Nathan,

We are going to do a force-pull and resolve those conflicts ourselves. No action needed on your part.

**Comment by NathanBaulch** [comment]
No worries @tonyliaoss, let me know if you need anything.

File: src/google/protobuf/descriptor.proto (hunk 1 above)
Comment (NathanBaulch) [reply src/google/protobuf/descriptor.proto:83]: Ah, this particular typo is what started this whole process for me! It leaks into downstream code that references `google/api/annotations.proto`, typically to annotate services with HTTP routing metadata.
If someone internal could "independently" fix this typo then that'd be great! 🙏

**Pull request closed without merging**